genic/main.c

```diff
@@ -123,32 +127,30 @@ int main(int argc, char **argv)
       myfree(seedtable);
   }
 
-  write_particle_data(1, &bf, 0, All2.Ngrid);
-  free_ffts();
+  write_particle_data(1, &bf, 0, All2.Ngrid, ICP, NumPart);
 
   /*Now make the gas if required*/
   if(All2.ProduceGas) {
 
-    if(!All2.MakeGlass) {
-        setup_grid(shift_gas, All2.Ngrid);
+    if(!All2.MakeGlassBar) {
```

Comment:
You sure want to go with Bar not Gas here?

Reply:
You're right! I want to go with Gas